src/rules.js

```diff
@@ -12,6 +12,20 @@ export function OperationsMustHaveNames(context) {
   };
 }
 
+function getFieldWasRequestedOnNode(node, field, recursing = false) {
+  return node.selectionSet.selections.some(n => {
+    // If it's an inline fragment, we need to look deeper
+    if (n.kind === 'InlineFragment' && !recursing) {
```

Comment:
i actually wonder if we _do_ need to keep recursing — is it possible to have inline fragments inside inline fragments?

i'll leave this for now since this unblocks some breaking behavior